### PR TITLE
Admin user management flow/ email updates

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add various new email templating and workers needed for the admin user management flow updates

## WHY
we will need to send this emails as part of the new signup flow for admin users

## HOW
add a series of new templates, works and methods on the `User` model to execute these emails. Please note: there is some shared content between these emails, but I opted to keep most of these separate as the content may change in the future. The logic for when to send each email should stay the same though. 
### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

File: services/QuillLMS/app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -40,6 +45,99 @@ def account_created_email(user, temp_password, admin_name)
...
+  end
+
+  def internal_tool_made_district_admin_email(user, district_name)
+    @user = user
+    @district_name = district_name
+    @set_password_link = ENV['DEFAULT_URL']
+    mail to: user.email, subject: "#{user.first_name}, you are now a Quill admin for #{district_name}"
+  end
+
   # Sent when an admin requests an existing teacher to join one of their schools.
```
Comment (emilia-friedberg) [inline services/QuillLMS/app/mailers/user_mailer.rb]: This looks like it's going to be a duplicate of one of the emails above, is this just gonna get cleaned up later?

**Review by emilia-friedberg — APPROVED** [review]
This seems soooooo tedious. Thank you for doing all of that. The code looks good to me in general but I wonder if, since it's just so many functions and files, it makes sense to group the individual worker files into a `workers` subfolder (`admin_user_management_email_workers` or something), and same with the `user_mailer` erb files. And then maybe move all those functions in `user` into a helper module. That's just organizational preference but I'm worried all the new code will make it harder to scan those folders and files quickly when you're looking for something.

**Comment by brendanshean** [comment]
I was actually thinking the same thing @emilia-friedberg said about reorganization.  without knowing too much about the code it looks like there might be two folders naturally:
Screenshot 2022-12-02 at 8 10 14 AM

File: services/QuillLMS/app/mailers/user_mailer.rb (hunk 1 above)
Comment (eadams17) [reply services/QuillLMS/app/mailers/user_mailer.rb]: yeah so this is part of the existing signup flow-- this email will get removed once the last two cards are implemented and these new emails are used instead

File: services/QuillLMS/spec/models/user_spec.rb [hunk 2]
```diff
@@ -57,7 +57,7 @@
 #
 require 'rails_helper'
 
-describe User, type: :model do
+describe User, type: :model do # rubocop:disable Metrics/BlockLength
```
Comment (brendanshean) [inline services/QuillLMS/spec/models/user_spec.rb]: Usually a good indicator to break up underlying class into new classes.

**Review by brendanshean — APPROVED** [review]
Other than reorganizing folder structure, I think this looks great and is ready to go.

**Pull request merged**